vcpkg/users/github-dependency-graph.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Integrating with GitHub dependency graph
+description: Record the set of resolved libraries with the GitHub dependency graph
+ms.date: 05/22/2023
+---
+
+# Integrating with the GitHub dependency graph
+
+> [!TIP]
+> This page describes an experimental feature of vcpkg. We are actively developing this feature and would like to hear your feedback. Let us know if you have any thoughts about the current functionality or any desired functionality by emailing vcpkg@microsoft.com or by filing an issue in [vcpkg-docs](https://github.com/microsoft/vcpkg-docs/issues). If you have feedback on any of the GitHub features that this enables, let us know tha too and we'll make sure it gets to the right folks at GitHub.
+
+## About the GitHub dependency graph
+
+The GitHub [dependency graph](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-the-dependency-graph) stores the set of dependencies for a repository. Beyond just being able to visualize what a repository's dependencies are, GitHub builds several useful features on top of this data, including [dependency review](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-dependency-review) and [Dependabot alerts](https://docs.github.com/en/code-security/dependabot/dependabot-alerts/about-dependabot-alerts).
+
+## vcpkg integration with the GitHub depdency graph
+
+vcpkg has experimental support for populating the GitHub dependency graph from within a GitHub Action workflow. In order to enable this feature, you must include the following in your workflow file:
+
+* Set the `dependencygraph` feature flag to the environment variable `VCPKG_FEATURE_FLAGS`.
+* Set the `GITHUB_TOKEN` environment variable with the GitHub workflow variable `${{ secrets.GITHUB_TOKEN }}`.
+* Enable write access to the contents of your repository by including the following permissions block. This is because it must write dependency graph metadata to your repository. This __will not__ add any commits to your repository or otherwise modify the contents of your repository (despite the permission name).
```

Comment:
Perhaps we should add a section here asking users to enable dependency graph support for their repo

Reply:
I think that's a good idea. If it's a public repository, it's enabled by default, but if it's not, then we should give them an idea of how to do it. Maybe linking to some GitHub doc would be sufficient. I'll see what makes most sense (I know I saw somewhere someone trying to invent a button that could easily do it, but I think the GH settings UI has improved since that time).